one more deprecated g_type_class_add_private occurrence in the unit tests

Follow-up issue for #10 and #14:

```
[ 62%] Building C object tests/CMakeFiles/backendmock.dir/backend-mock-guest.c.o
cd /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session/obj-x86_64-linux-gnu/tests && /usr/bin/cc -DINDICATOR_SERVICE_DIR=\"/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session/obj-x86_64-linux-gnu/tests\" -DSCHEMA_DIR=\"/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session/obj-x86_64-linux-gnu/tests\" -I/usr/include/gio-unix-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/uuid -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session/src -I/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session/obj-x86_64-linux-gnu/src -I/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session/obj-x86_64-linux-gnu/tests  -g -O2 -fdebug-prefix-map=/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2     -Wall -pedantic -Wextra -Wno-missing-field-initializers -g -o CMakeFiles/backendmock.dir/backend-mock-guest.c.o   -c /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session/tests/backend-mock-guest.c
/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session/tests/backend-mock-guest.c: In function ‘indicator_session_guest_mock_class_init’:
/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session/tests/backend-mock-guest.c:104:3: warning: ‘g_type_class_add_private’ is deprecated [-Wdeprecated-declarations]
   g_type_class_add_private (klass, sizeof (IndicatorSessionGuestMockPriv));
   ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session/tests/backend-mock-guest.h:24,
                 from /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session/tests/backend-mock-guest.c:22:
/usr/include/glib-2.0/gobject/gtype.h:1303:10: note: declared here
 void     g_type_class_add_private       (gpointer                    g_class,
          ^~~~~~~~~~~~~~~~~~~~~~~~
[ 63%] Building C object tests/CMakeFiles/backendmock.dir/backend-mock-users.c.o
cd /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session/obj-x86_64-linux-gnu/tests && /usr/bin/cc -DINDICATOR_SERVICE_DIR=\"/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session/obj-x86_64-linux-gnu/tests\" -DSCHEMA_DIR=\"/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session/obj-x86_64-linux-gnu/tests\" -I/usr/include/gio-unix-2.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/uuid -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session/src -I/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session/obj-x86_64-linux-gnu/src -I/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session/obj-x86_64-linux-gnu/tests  -g -O2 -fdebug-prefix-map=/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2     -Wall -pedantic -Wextra -Wno-missing-field-initializers -g -o CMakeFiles/backendmock.dir/backend-mock-users.c.o   -c /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session/tests/backend-mock-users.c
/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session/tests/backend-mock-users.c: In function ‘indicator_session_users_mock_class_init’:
/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session/tests/backend-mock-users.c:124:3: warning: ‘g_type_class_add_private’ is deprecated [-Wdeprecated-declarations]
   g_type_class_add_private (klass, sizeof (IndicatorSessionUsersMockPriv));
   ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:23,
                 from /usr/include/glib-2.0/gio/gioenums.h:28,
                 from /usr/include/glib-2.0/gio/giotypes.h:28,
                 from /usr/include/glib-2.0/gio/gio.h:26,
                 from /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session/tests/backend-mock.h:23,
                 from /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-session.upstream/ayatana-indicator-session/tests/backend-mock-users.c:20:
/usr/include/glib-2.0/gobject/gtype.h:1303:10: note: declared here
 void     g_type_class_add_private       (gpointer                    g_class,
          ^~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/tests/backend-mock-guest.c
+++ b/tests/backend-mock-guest.c
@@ -21,18 +21,16 @@
 
 #include "backend-mock-guest.h"
 
-struct _IndicatorSessionGuestMockPriv
+struct _IndicatorSessionGuestMockPrivate
 {
   gboolean guest_is_active;
   gboolean guest_is_logged_in;
   gboolean guest_is_allowed;
 };
 
-typedef IndicatorSessionGuestMockPriv priv_t;
+typedef IndicatorSessionGuestMockPrivate priv_t;
 
-G_DEFINE_TYPE (IndicatorSessionGuestMock,
-               indicator_session_guest_mock,
-               INDICATOR_TYPE_SESSION_GUEST)
+G_DEFINE_TYPE_WITH_PRIVATE(IndicatorSessionGuestMock, indicator_session_guest_mock, INDICATOR_TYPE_SESSION_GUEST)
 
 /**
 ***  IndicatorSessionGuest virtual functions
@@ -100,8 +98,6 @@ indicator_session_guest_mock_class_init (IndicatorSessionGuestMockClass * klass)
   guest_class->is_logged_in = my_is_logged_in;
   guest_class->is_active = my_is_active;
   guest_class->switch_to_guest = my_switch_to_guest;
-
-  g_type_class_add_private (klass, sizeof (IndicatorSessionGuestMockPriv));
 }
 
 static void
@@ -110,9 +106,7 @@ indicator_session_guest_mock_init (IndicatorSessionGuestMock * self)
 {
   priv_t * p;
 
-  p = G_TYPE_INSTANCE_GET_PRIVATE (self,
-                                   INDICATOR_TYPE_SESSION_GUEST_MOCK,
-                                   IndicatorSessionGuestMockPriv);
+  p = indicator_session_guest_mock_get_instance_private(self);
   self->priv = p;
 
   p->guest_is_allowed = TRUE;

--- a/tests/backend-mock-guest.h
+++ b/tests/backend-mock-guest.h
@@ -33,7 +33,7 @@ G_BEGIN_DECLS
 #define INDICATOR_IS_SESSION_GUEST_MOCK(o)         (G_TYPE_CHECK_INSTANCE_TYPE ((o), INDICATOR_TYPE_SESSION_GUEST_MOCK))
 
 typedef struct _IndicatorSessionGuestMock        IndicatorSessionGuestMock;
-typedef struct _IndicatorSessionGuestMockPriv    IndicatorSessionGuestMockPriv;
+typedef struct _IndicatorSessionGuestMockPrivate IndicatorSessionGuestMockPrivate;
 typedef struct _IndicatorSessionGuestMockClass   IndicatorSessionGuestMockClass;
 
 /**
@@ -43,7 +43,7 @@ struct _IndicatorSessionGuestMock
 {
   /*< private >*/
   IndicatorSessionGuest parent;
-  IndicatorSessionGuestMockPriv * priv;
+  IndicatorSessionGuestMockPrivate * priv;
 };
 
 struct _IndicatorSessionGuestMockClass

--- a/tests/backend-mock-users.c
+++ b/tests/backend-mock-users.c
@@ -20,16 +20,14 @@
 #include "backend-mock.h"
 #include "backend-mock-users.h"
 
-struct _IndicatorSessionUsersMockPriv
+struct _IndicatorSessionUsersMockPrivate
 {
   GHashTable * users;
 };
 
-typedef IndicatorSessionUsersMockPriv priv_t;
+typedef IndicatorSessionUsersMockPrivate priv_t;
 
-G_DEFINE_TYPE (IndicatorSessionUsersMock,
-               indicator_session_users_mock,
-               INDICATOR_TYPE_SESSION_USERS)
+G_DEFINE_TYPE_WITH_PRIVATE(IndicatorSessionUsersMock, indicator_session_users_mock, INDICATOR_TYPE_SESSION_USERS)
 
 /***
 ****  IndicatorSessionUsers virtual functions
@@ -120,8 +118,6 @@ indicator_session_users_mock_class_init (IndicatorSessionUsersMockClass * klass)
   users_class->get_uids = my_get_uids;
   users_class->get_user = my_get_user;
   users_class->activate_user = my_activate_user;
-
-  g_type_class_add_private (klass, sizeof (IndicatorSessionUsersMockPriv));
 }
 
 static void
@@ -130,9 +126,7 @@ indicator_session_users_mock_init (IndicatorSessionUsersMock * self)
 {
   priv_t * p;
 
-  p = G_TYPE_INSTANCE_GET_PRIVATE (self,
-                                   INDICATOR_TYPE_SESSION_USERS_MOCK,
-                                   IndicatorSessionUsersMockPriv);
+  p = indicator_session_users_mock_get_instance_private(self);
   self->priv = p;
 
   p->users = g_hash_table_new_full (g_direct_hash,

--- a/tests/backend-mock-users.h
+++ b/tests/backend-mock-users.h
@@ -33,7 +33,7 @@ G_BEGIN_DECLS
 #define INDICATOR_IS_SESSION_USERS_MOCK(o)         (G_TYPE_CHECK_INSTANCE_TYPE ((o), INDICATOR_TYPE_SESSION_USERS_MOCK))
 
 typedef struct _IndicatorSessionUsersMock        IndicatorSessionUsersMock;
-typedef struct _IndicatorSessionUsersMockPriv    IndicatorSessionUsersMockPriv;
+typedef struct _IndicatorSessionUsersMockPrivate IndicatorSessionUsersMockPrivate;
 typedef struct _IndicatorSessionUsersMockClass   IndicatorSessionUsersMockClass;
 
 /**
@@ -43,7 +43,7 @@ struct _IndicatorSessionUsersMock
 {
   /*< private >*/
   IndicatorSessionUsers parent;
-  IndicatorSessionUsersMockPriv * priv;
+  IndicatorSessionUsersMockPrivate * priv;
 };
 
 struct _IndicatorSessionUsersMockClass

--- a/tests/test-service.cc
+++ b/tests/test-service.cc
@@ -24,6 +24,13 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "backend-mock-guest.h"
 #include "backend-mock-actions.h"
 
+gboolean onMainLoopQuit(gpointer pUserData)
+{
+    g_main_loop_quit((GMainLoop*)pUserData);
+
+    return FALSE;
+}
+
 /***
 ****
 ***/
@@ -169,7 +176,7 @@ class ServiceTest: public GTestDBusFixture
                                                              G_BUS_NAME_WATCHER_FLAGS_NONE,
                                                              on_name_appeared, // quits the loop
                                                              NULL, this, NULL);
-      const guint timer_id = g_timeout_add_seconds (TIME_LIMIT_SEC, (GSourceFunc)g_main_loop_quit, loop);
+      const guint timer_id = g_timeout_add_seconds(TIME_LIMIT_SEC, onMainLoopQuit, loop);
       g_main_loop_run (loop);
       g_source_remove (timer_id);
       g_bus_unwatch_name (watch_id);

--- a/tests/test-service.cc
+++ b/tests/test-service.cc
@@ -4,16 +4,16 @@ Copyright 2012 Canonical Ltd.
 Authors:
     Charles Kerr <charles.kerr@canonical.com>
 
-This program is free software: you can redistribute it and/or modify it 
-under the terms of the GNU General Public License version 3, as published 
+This program is free software: you can redistribute it and/or modify it
+under the terms of the GNU General Public License version 3, as published
 by the Free Software Foundation.
 
-This program is distributed in the hope that it will be useful, but 
-WITHOUT ANY WARRANTY; without even the implied warranties of 
-MERCHANTABILITY, SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR 
+This program is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranties of
+MERCHANTABILITY, SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR
 PURPOSE.  See the GNU General Public License for more details.
 
-You should have received a copy of the GNU General Public License along 
+You should have received a copy of the GNU General Public License along
 with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 

